src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -56,14 +56,23 @@ private BoundDeconstructionAssignmentOperator BindDeconstructionAssignment(CShar
 
             if ((object)boundRHS.Type == null)
```

Comment:
Do we have a good rule for when errors will cause `Type` to be `ErrorType` and when it will be `null`?  Having both without a justification for when to choose one or the other make it harder to review code.